timetracker/entries/views.py

```diff
@@ -1,113 +1,104 @@
-            entry = Entry()
-            entry.start = entry_form.cleaned_data['start']
-            entry.stop = entry_form.cleaned_data['stop']
-            entry.project = entry_form.cleaned_data['project']
-            entry.description = entry_form.cleaned_data['description']
-            entry.save()
-            return redirect('entry-list')
-    else:
-        entry_form = EntryForm()
-
-    entry_list = Entry.objects.all()
-    return render(request, 'entries.html', {
-        'entry_list': entry_list,
-        'entry_form': entry_form,
-    })
-
-
-def projects(request):
-    if request.method == 'POST':
-        # Create our form object with our POST data
-        form = ProjectForm(request.POST)
-        if form.is_valid():
-            Project.objects.create(
-                name=form.cleaned_data['name'],
-                client=form.cleaned_data['client']
-            )
-            return redirect('project-list')
-    else:
-        form = ProjectForm()
-
-    project_list = Project.objects.all()
-    return render(request, 'projects.html', {
-        'project_list': project_list,
-        'form': form
-    })
-
-
-def project_detail(request, pk):
-    project = get_object_or_404(Project, pk=pk)
-
-    if request.method == 'POST':
-        form = ProjectForm(request.POST)
-        if form.is_valid():
-            # Update project details
-            project.name = form.cleaned_data['name']
-            project.client=form.cleaned_data['client']
-            project.save()
-            return redirect('project-list')
-    else:
-        # Initialise form with project data
-        form = ProjectForm(initial={'name': project.name, 'client': project.client})
-
-    project = get_object_or_404(Project, pk=pk)
-    return render(request, 'project_detail.html', {
-        'project': project,
-        'form': form,
-    })
+class ClientCreateView(CreateView):
+    model = Client
+    template_name = 'clients.html'
+    form_class = ClientForm
+    context_object_name = 'client_list'
+    success_url = reverse_lazy('client-list')
+    test = Client.objects.all()
+
+    def get_context_data(self, **kwargs):
+        context = super(ClientCreateView, self).get_context_data(**kwargs)
+        context['client_list'] = self.model.objects.all()
+        return context
+
+
+class ClientUpdate(UpdateView):
+    model = Client
+    template_name='client_detail.html'
+    form_class = ClientForm
+    context_object_name = 'client_list'
+    success_url = reverse_lazy('client-list')
+
+    def get_object(self, queryset=None):
+        if queryset is None:
+            queryset = self.get_queryset()
+
+        pk = self.kwargs.get(self.pk_url_kwarg, None)
+
+        if pk is not None:
+            queryset = queryset.filter(pk=pk)
+
+        try:
+            obj = queryset.get()
+
+        except queryset.model.DoesNotExist:
+            raise Http404("Client id doesn't exist")
+
+        return obj
```

Comment:
I didn't knew that UpdateView would magically populated the form for requested Client pk so therefore in order to achieve this functionality I used the method get_object which is referenced here https://ccbv.co.uk/projects/Django/1.8/django.views.generic.edit/UpdateView/